### PR TITLE
Delete editor site launch

### PR DESCRIPTION
**Pull request by noahtallen**
Related to #

## Proposed Changes
I couldn't find anywhere this directory was imported, so I'm deleting it (and also removing some unused deps in ETK). Related partially to #74551.

For this feature to load, its PHP would have to be enqueued by something.... and I don't think it is 😁 

I accidentally discovered this today, but when looking back, I found we've discussed this previously: #53118

## Testing Instructions
1. Sync ETK to your sandbox and verify nothing crashes (including checking PHP err logs)
2. See if you can find anywhere this directory is imported.

### CODE REVIEW

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts [hunk 1]
```diff
@@ -1,102 +0,0 @@
-import { recordTracksEvent } from '@automattic/calypso-analytics';
-import { dispatch } from '@wordpress/data';
-import domReady from '@wordpress/dom-ready';
-import { __ } from '@wordpress/i18n';
-import { inIframe } from '../../../block-inserter-modifications/contextual-tips/utils';
-import { GUTENBOARDING_LAUNCH_FLOW } from '../constants';
-
-import 'a8c-fse-common-data-stores';
-import '@wordpress/editor';
```
Comment (tyxla) [inline apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts:9]: Is this removing the last use of `@wordpress/editor`? Can we remove that dependency from editing toolkit altogether?

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/styles.scss [hunk 2]
```diff
@@ -1,21 +0,0 @@
...
-			box-shadow: none !important;
-
-			&[aria-disabled="true"] {
-				opacity: 0.3 !important;
-			}
-		}
-	}
-}
-
-.editor-gutenberg-launch__launch-button {
```
Comment (tyxla) [inline apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/styles.scss:18]: We can also likely delete this then:

https://github.com/Automattic/wp-calypso/blob/b7a4a072ae2a13785fb6ccbf9c34ed8e65739705/packages/launch/src/focused-launch/style.scss#L7-L9

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-steps/plan-step/styles.scss [hunk 3]
```diff
@@ -1,20 +0,0 @@
-@import "@automattic/onboarding/styles/mixins";
-@import "@automattic/plans-grid/src/variables";
```
Comment (tyxla) [inline apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-steps/plan-step/styles.scss:2]: Seems like we can remove the `@automattic/plans-grid` dependency from ETK now.

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-steps/plan-step/index.tsx [hunk 4]
```diff
@@ -1,86 +0,0 @@
-import { useSite } from '@automattic/launch';
```
Comment (tyxla) [inline apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-steps/plan-step/index.tsx:1]: Seems like we can remove the `@automattic/launch` dependency from ETK now.

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-steps/final-step/index.tsx [hunk 5]
```diff
@@ -1,282 +0,0 @@
-import {
-	CheckoutStepBody,
-	checkoutTheme,
-	CheckoutSummaryArea,
-	CheckoutSummaryCard,
-	MainContentWrapper,
-	CheckoutStepAreaWrapper,
-	SubmitButtonWrapper,
-	FormStatus,
-} from '@automattic/composite-checkout';
```
Comment (tyxla) [inline apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-steps/final-step/index.tsx:10]: Seems like we can remove the `@automattic/composite-checkout` dependency from ETK now.

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-steps/final-step/index.tsx [hunk 6]
```diff
@@ -1,282 +0,0 @@
...
-import { useLocale, useLocalizeUrl } from '@automattic/i18n-utils';
-import { useSiteDomains, useDomainSuggestion, useDomainSearch, useTitle } from '@automattic/launch';
-import { Title, SubTitle, ActionButtons, BackButton } from '@automattic/onboarding';
-import { ThemeProvider } from '@emotion/react';
```
Comment (tyxla) [inline apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-steps/final-step/index.tsx:14]: Do we still need `@emotion/react` as a dependency of ETK after the removal of this? I think we don't.

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-steps/final-step/index.tsx [hunk 7]
```diff
@@ -1,282 +0,0 @@
...
-import { Button, Tip } from '@wordpress/components';
-import { useDispatch, useSelect } from '@wordpress/data';
-import { createInterpolateElement } from '@wordpress/element';
-import { sprintf } from '@wordpress/i18n';
-import { Icon, check } from '@wordpress/icons';
-import { useI18n } from '@wordpress/react-i18n';
```
Comment (tyxla) [inline apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-steps/final-step/index.tsx:20]: Seems like we no longer use `@wordpress/react-i18n` after this PR. Should we remove that dependency and its special handling here: 

https://github.com/Automattic/wp-calypso/blob/a14b6727ffc1ed68b951c95f8a43ca6e1cb15ccc/apps/editing-toolkit/webpack.config.js#L107

**Review by tyxla — APPROVED** [review]
Great work @noahtallen!

I've suggested removing a few more dependencies that we no longer seem to use.

Could be done in a separate PR if that works better for you.

🚀

**Comment by noahtallen** [comment]
Good finds! I'll look into removing those in this PR.

**Comment by tyxla** [comment]
I'm going to proceed and merge this to unblock some further cleanup work.

Going to open a new PR with my suggestions so we don't lose track of them. 

Thanks, @noahtallen! 🚀

**Comment by tyxla** [comment]
Follow-up cleanup: #75267

**Comment by noahtallen** [comment]
Thanks!

**Pull request merged**